doc/source/contributing_docstring.rst

```diff
@@ -0,0 +1,634 @@
+======================
+pandas docstring guide
+======================
+
+About docstrings and standards
+------------------------------
+
+A Python docstring is a string used to document a Python function or method,
+so programmers can understand what it does without having to read the details
+of the implementation.
+
+Also, it is a commonn practice to generate online (html) documentation
+automatically from docstrings. `Sphinx <http://www.sphinx-doc.org>`_ serves
+this purpose.
+
+Next example gives an idea on how a docstring looks like:
+
+.. code-block:: python
+
+    def add(num1, num2):
+    """Add up two integer numbers.
+
+    This function simply wraps the `+` operator, and does not
+    do anything interesting, except for illustrating what is
+    the docstring of a very simple function.
+
+    Parameters
+    ----------
+    num1 : int
+        First number to add
+    num2 : int
+        Second number to add
+
+    Returns
+    -------
+    int
+        The sum of `num1` and `num2`
+
+    Examples
+    --------
+    >>> add(2, 2)
+    4
+    >>> add(25, 0)
+    25
+    >>> add(10, -10)
+    0
+    """
+    return num1 + num2
+
+Some standards exist about docstrings, so they are easier to read, and they can
+be exported to other formats such as html or pdf.
+
+The first conventions every Python docstring should follow are defined in
+`PEP-257 <https://www.python.org/dev/peps/pep-0257/>`_.
+
+As PEP-257 is quite open, and some other standards exist on top of it. In the
+case of pandas, the numpy docstring convention is followed. There are two main
+documents that explain this convention:
+
+- `Guide to NumPy/SciPy documentation <https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt>`_
+- `numpydoc docstring guide <http://numpydoc.readthedocs.io/en/latest/format.html>`_
```

Comment:
Even if the last one should contain the same as the first, but with a better presentation, I think it's good that people is aware of the first document. People is not really expected to read or follow them, they're presented here just for reference. I'll leave both unless you really feel we should get rid of the first.

Reply:
I personally think it will just confuse people by giving 2 links. If there are two, I expect that somehow it is useful that I look at both of them, but then wonder what is the different, because the content is almost *exactly* the same.

Reply:
That's a good point. I left just the numpy doc one in the list, and kept the other just in a comment. Let me know if you think it's still worthless to have it this way.